### ISSUE TITLE
translate.py - empty _buckets sets will crash

### ISSUE THREAD

**Issue by marcotrombetti** [tensorflow/tensorflow#649]
If the dev_set does not contain all the string related to all the _buckets 
the script will crash.
Eg. If the dev set does not contain a string between 20 and 25 words, the script will crash.

translate.py, around line 178

``` python
encoder_inputs, decoder_inputs, target_weights = model.get_batch(
              dev_set, bucket_id)
...
```

I think that before that line you should had add an if to avoid that

``` python
     if len(dev_set[bucket_id])>0:
           encoder_inputs, decoder_inputs, target_weights = model.get_batch(
              dev_set, bucket_id)
           _, eval_loss, _ = model.step(sess, encoder_inputs, decoder_inputs,
                                       target_weights, bucket_id, True)
           eval_ppx = math.exp(eval_loss) if eval_loss < 300 else float('inf')
           print("  eval: bucket %d perplexity %.2f" % (bucket_id, eval_px))
```

**Comment by keveman** [comment]
marcotrombetti@, please do send a PR with a fix.

**Comment by ybbaigo** [comment]
encounter same issue as @marcotrombetti , I just send a PR with fix @keveman #1496

**Comment by itsmeolivia** [comment]
looks like the PR fixed this!
